cineast-core/src/main/java/org/vitrivr/cineast/core/features/abstracts/AbstractSegmentExporter.java

```diff
@@ -0,0 +1,124 @@
+package org.vitrivr.cineast.core.features.abstracts;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.vitrivr.cineast.core.data.segments.SegmentContainer;
+import org.vitrivr.cineast.core.db.PersistencyWriterSupplier;
+import org.vitrivr.cineast.core.db.setup.EntityCreator;
+import org.vitrivr.cineast.core.features.extractor.Extractor;
+import org.vitrivr.cineast.core.util.LogHelper;
+
+abstract public class AbstractSegmentExporter implements Extractor {
+
+  protected static final Logger LOGGER = LogManager.getLogger();
+
+  private static final String PROPERTY_NAME_DESTINATION = "destination";
+
+  /**
+   * Destination path for the audio-segment.
+   */
+  private final Path destination;
```

Comment:
Why is this private object property documented and the following protected properties aren't?
I heavily suggest to document important information particularly on things that are exposed.
Our software stack has certainly room for improvement in terms of internal and external documentation and going forward, it seems to me it would be a good idea to aim for better documentation...

Reply:
I assume the path should be a folder? The documentation could be more precise.